colossalai/fx/proxy.py

```diff
@@ -1,12 +1,39 @@
 import operator
+import uuid
+from typing import Any, List, Tuple, Union
+
 import torch
-from torch.fx.proxy import Proxy, Attribute
-from typing import List, Union, Any
+from torch.fx.proxy import Attribute, Proxy
+
 from colossalai.fx.tracer.meta_patch import meta_patched_function
 
 __all__ = ['ColoProxy']
 
 
+def _set_data_ptr(data: Union[torch.Tensor, List, Tuple]) -> None:
+    """
+    Set a non-trivial data_ptr for the data if it is a ``torch.Tensor`` and is on ``meta`` device
+    """
+    if isinstance(data, list) or isinstance(data, tuple) and not isinstance(data, torch.Size):
+        for item in data:
+            _set_data_ptr(item)
+    elif isinstance(data, torch.Tensor):
+        if not data.data_ptr():
+            data_ptr = uuid.uuid4()
+            data.data_ptr = lambda: data_ptr
```

Comment:
Use PyTree instead to set data pointer instead as different data structures can be expected.